subprojects/platform-jvm/src/main/java/org/gradle/api/tasks/WorkerProcessTask.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.provider.Property;
+import org.gradle.jvm.toolchain.JavaLauncher;
+
+/**
+ * A {@code WorkerProcessTask} is a task which starts one (or more) worker processes during execution.
+ *
+ * @since 8.4
+ */
+@Incubating
+public interface WorkerProcessTask {
```

Comment:
❓ We still have the issue that you can't do this in Java:
```
tasks.withType(WorkerProcessTask.class).configureEach {
}
```

Do we need to make `WorkerProcessTask` extend `Task`?

Reply:
💭 If this worked, we could do something like this in `BasePlugin`:
```
tasks.withType(WorkerProcessTask.class).configureEach { task ->
    task.getMaxHeapSize().convention("512m");
}
```
This would make the convention of max heap size visible and applied consistently.